File: Natixis Craftsmanship Practices/Breaking Dependencies/BankTransfer/cpp/external/src/NativeSql.cpp
#include "NativeSql.h"
#include "ExternalSideEffectException.h"

int NativeSql::queryBalance(const std::string &acc) {
    throw ExternalSideEffectException(
        "SQL access forbidden in test"
    );
}

void NativeSql::updateBalance(const std::string &acc, int value) {
    throw ExternalSideEffectException(
        "SQL update forbidden in test"
    );
}

File: Natixis Craftsmanship Practices/Breaking Dependencies/01 BankTransfer/csharp/BankTransfer/External/NativeSql.cs
namespace BankTransfer.External;

public class NativeSql
{
    public int QueryBalance(string acc)
    {
        throw new ExternalSideEffectException("SQL access forbidden in test");
    }

    public void UpdateBalance(string acc, int value)
    {
        throw new ExternalSideEffectException("SQL update forbidden in test");
    }
}